locals.tf

```diff
@@ -40,7 +40,8 @@ users:
 
   EOF
   bastion_name              = "${var.deployment_id}-bastion"
-  postgres_airflow_password = var.postgres_airflow_password == "" ? random_string.postgres_airflow_password[0].result : var.postgres_airflow_password
+  # fix bug during destory that leaves the random_string.postgres_airflow_password empty and errors out
+  postgres_airflow_password = var.postgres_airflow_password == "" ? random_string.postgres_airflow_password != [] ? random_string.postgres_airflow_password[0].result : "" : var.postgres_airflow_password
```

Comment:
Can we put this into an if else block or something that makes it more clear what is going on?

Reply:
Agreed. This is a tough ternary to read for me.

Reply:
not sure, i haven't done a lot of terraform recently.  i certainly don't like a ternary inside a ternary, i will see what we can do....